### PR TITLE
add more tests, and some placeholders

### PR DESCRIPTION
**Pull request by Tieske**
More test cases (GPT generated mostly)

### CODE REVIEW

File: homie5/formats/datetime.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+description: validating dateTime formats
+
+# This type does not have a specified format entry in the Homie convention
```
Comment (nstadigs) [inline homie5/formats/datetime.yml:3]: Is there any value to test whether a format is provided anyway?

File: homie5/formats/duration.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+description: validating duration formats
+
+# This type does not have a specified format entry in the Homie convention
```
Comment (nstadigs) [inline homie5/formats/duration.yml:3]: Is there any value to test whether a format is provided anyway?

File: homie5/formats/string.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+description: validating string formats
+
+# This type does not have a specified format entry in the Homie convention
```
Comment (nstadigs) [inline homie5/formats/string.yml:3]: Is there any value to test whether a format is provided anyway?

**Review by nstadigs — COMMENTED** [review]
I checked the tests against the convention and I think they are correct. Left a comment in a few places about tests for types that doesn't have a specified format, but I'm unsure if implementations should just ignore the format or error early.

File: homie5/formats/datetime.yml (hunk 1 above)
Comment (Tieske) [reply homie5/formats/datetime.yml:3]: good call. I don't think the spec currently forbids to pass in a format property for these types. Should we add that?
wdyt @schaze ?

Comment (nstadigs) [reply homie5/formats/datetime.yml:3]: From the spec (about forward compatibility):

> Controllers should ignore unknown fields, properties, attributes, etc. within an object (device, node, or property), but keep the object itself.
> Controllers should ignore the entire object (device, node, or property) if in a known field, property, or >attribute an illegal value is encountered. For example;
>
>    - illegal characters in a topic or name
>    - unknown data type
>    - unknown/illegal format
>    - required element missing

But it's not really an "unknown" field, right? It just doesn't have a format... :shrug:

Comment (nstadigs) [reply homie5/formats/datetime.yml:3]: oh, but this is for controllers. Not devices... makes sense. Sorry, I'm new to this :sweat_smile:

Comment (Tieske) [reply homie5/formats/datetime.yml:3]: But forward compatibility may add it in the future. So maybe we should leave it as is, and just ignore it.

**Review by nstadigs — APPROVED** [review]

**Pull request merged**